lib/providers/config_provider.dart

```diff
@@ -1,142 +1,370 @@
+          .where((widget) =>
+              board.widgets.contains(widget.id) &&
+              quarantineWidgets.indexWhere((element) => element.id == widget.id) == -1 &&
+              widget.type != "WhiteSpaceWidget")
+          .toList()
+    ];
+    return getSortedWidgetsList(boardWidgets);
   }
 
-  void updateWidget(Map<String, dynamic> widgetData) {
-    this._config.widgets.widgetsById.forEach((key, value) {
-      if (key == widgetData['id']) {
-        value.updateWidget(widgetData);
-        notifyListeners();
+  List<DashboardWidget> getSortedWidgetsList(List<DashboardWidget> boardWidgets) {
+    switch (_settingsPreferences.sortBy) {
+      case WidgetSortTypes.NONE:
+        return boardWidgets;
+      case WidgetSortTypes.NAME_ASCENDING:
+        boardWidgets.sort((a, b) => getWidgetName(a).compareTo(getWidgetName(b)));
+        break;
+      case WidgetSortTypes.NAME_DESCENDING:
+        boardWidgets.sort((a, b) => getWidgetName(a).compareTo(getWidgetName(b)));
+        boardWidgets = boardWidgets.reversed.toList();
+        break;
+      case WidgetSortTypes.STATUS_ASCENDING:
+        boardWidgets.sort((a, b) => sortWidgetsByStatus(a, b));
+        break;
+      case WidgetSortTypes.STATUS_DESCENDING:
+        boardWidgets.sort((a, b) => sortWidgetsByStatus(a, b));
+        boardWidgets = boardWidgets.reversed.toList();
+        break;
+    }
+    return boardWidgets;
+  }
+
+  int sortWidgetsByStatus(DashboardWidget a, DashboardWidget b) {
+    int firstWidgetStatusSortValue = widgetStatusSortValue(a);
+    int secondWidgetStatusSortValue = widgetStatusSortValue(b);
+    if (firstWidgetStatusSortValue == secondWidgetStatusSortValue) {
+      return getWidgetName(a).compareTo(getWidgetName(b));
+    } else {
+      return firstWidgetStatusSortValue > secondWidgetStatusSortValue ? 1 : -1;
+    }
+  }
+
+  int widgetStatusSortValue(DashboardWidget widget) {
+    if (widget.content.containsKey(DashboardWidget.WIDGET_STATUS_KEY)) {
+      WidgetStatus widgetStatus =
+          EnumToString.fromString(WidgetStatus.values, widget.content[DashboardWidget.WIDGET_STATUS_KEY]);
+      switch (widgetStatus) {
+        case WidgetStatus.OK:
+          return 1;
+        case WidgetStatus.UNSTABLE:
+          return 5;
+        case WidgetStatus.FAIL:
+          return 6;
+        case WidgetStatus.UNKNOWN:
+          return 4;
+        case WidgetStatus.IN_PROGRESS:
+          return 2;
+        case WidgetStatus.ERROR_CONFIGURATION:
+          return 6;
+        case WidgetStatus.ERROR:
+          return 6;
+        case WidgetStatus.ERROR_CONNECTION:
+          return 6;
+        case WidgetStatus.TRANSPARENT:
+          return 2;
+        case WidgetStatus.CHECKBOX_OK:
+          return 1;
+        case WidgetStatus.CHECKBOX_FAIL:
+          return 6;
+        case WidgetStatus.CHECKBOX_UNKNOWN:
+          return 4;
+        case WidgetStatus.NONE:
+          return 3;
+        default:
+          return 3;
```

Comment:
**[LATER]** I wonder if those numbers 1, 2,3 ... could be a prop of a `WidgetStatus` class named `orderPriority`

Reply:
or we could just use index for it:
https://dart.dev/guides/language/language-tour#enumerated-types `WidgetStatus.CHECKBOX_FAIL.index`

Reply:
done